### DISCUSSION TITLE
Question about timers

### DISCUSSION THREAD

**Discussion by ghost** [elsa-workflows/elsa-core#1176] [Q&A] [answered]
I have the following definition in the designer =>
![image](https://user-images.githubusercontent.com/8503234/123411564-e0f44f80-d5b0-11eb-987d-3f39e1dd8ab1.png)

What I'm noticing is that after the send message the workflow is stopping. I expect it to continue sending messages indefinitely. Do we need to add a while loop here for this behavior? Because in the documents I do not see a while loop:
https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-recurring-tasks

**Comment by sfmskywalker** [comment, marked as answer]
The reason the workflow demonstrated in the recurring tasks guide executes repeatedly is because there it is used as a **start activity** (root activities are considered start activities). 

When the server starts, it collects all triggers by looking at the start activities of all workflows, which are used by a background task to repeatedly trigger the workflows associated with these triggers.

When you use the Timer activity as a non-root activity, it behaves slightly differently as you have observed; it now behaves like a one time delay.

In order for to repeatedly get this timer to trigger the workflow, you will indeed have to connect the Done outcome of the Send message activity to the Timer activity. You'll notice that now your workflow will never end.
